src/mlpack/core/data/string_encoding_policies/dictionary_encoding_policy.hpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file dictionary_encoding_policy.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of the DictionaryEncodingPolicy class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STRING_ENCODING_POLICIES_DICTIONARY_ENCODING_POLICY_HPP
+#define MLPACK_CORE_DATA_STRING_ENCODING_POLICIES_DICTIONARY_ENCODING_POLICY_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/data/string_encoding_policies/policy_traits.hpp>
+#include <mlpack/core/data/string_encoding.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * DicitonaryEnocdingPolicy is used as a helper class for StringEncoding.
+ * The encoder assigns a positive integer number to each unique token and treats
+ * the dataset as categorical. The numbers are assigned sequentially starting
+ * from one. The order in which the tokens are labeled is defined by
+ * the dictionary used by the StringEncoding class.
+ */
+class DictionaryEncodingPolicy
+{
+ public:
+  /**
+  * The function initializes the output matrix.
+  *
+  * @param output Output matrix to store the encoded results (sp_mat or mat).
+  * @param datasetSize The number of strings in the input dataset.
+  * @param maxNumTokens The maximum number of tokens in the strings of the 
+                        input dataset.
+  * @param dictionarySize The size of the dictionary (not used).
+  */
+  template<typename MatType>
+  static void InitMatrix(MatType& output,
```

Comment:
Would probably use ``const size_t`` just to be more clear that this parameter isn't going to be changed.

Reply:
Fixed everywhere.